File: src/Entity/Question.php
<?php
namespace App\Entity;

use App\Repository\QuestionRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: QuestionRepository::class)]
class Question
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Test::class, inversedBy: 'questions')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Test $test = null;

    #[ORM\Column(type: 'text')]
    private ?string $enonce = null;

    // On stocke 4 réponses
    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $answer1 = null;
    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $answer2 = null;
    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $answer3 = null;
    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $answer4 = null;

    // La bonne réponse (1, 2, 3 ou 4 par exemple)
    #[ORM\Column(type: 'integer')]
    private ?int $correctAnswer = null;

    // GETTERS / SETTERS ...
    public function getId(): ?int
    {
        return $this->id;
    }
    public function getTest(): ?Test
    {
        return $this->test;
    }
    public function setTest(?Test $test): self
    {
        $this->test = $test;
        return $this;
    }
    public function getEnonce(): ?string
    {
        return $this->enonce;
    }
    public function setEnonce(string $enonce): self
    {
        $this->enonce = $enonce;
        return $this;
    }
    public function getAnswer1(): ?string
    {
        return $this->answer1;
    }
    public function setAnswer1(?string $answer1): self
    {
        $this->answer1 = $answer1;
        return $this;
    }
    public function getAnswer2(): ?string
    {
        return $this->answer2;
    }
    public function setAnswer2(?string $answer2): self
    {
        $this->answer2 = $answer2;
        return $this;
    }
    public function getAnswer3(): ?string
    {
        return $this->answer3;
    }
    public function setAnswer3(?string $answer3): self
    {
        $this->answer3 = $answer3;
        return $this;
    }
    public function getAnswer4(): ?string
    {
        return $this->answer4;
    }
    public function setAnswer4(?string $answer4): self
    {
        $this->answer4 = $answer4;
        return $this;
    }
    public function getCorrectAnswer(): ?int
    {
        return $this->correctAnswer;
    }
    public function setCorrectAnswer(int $correctAnswer): self
    {
        $this->correctAnswer = $correctAnswer;
        return $this;
    }
}

File: Edu9raya-java/src/main/java/entities/Question.java
package entities;

public class Question {
    private int id;
    private int testId;
    private String enonce;
    private String answer1;
    private String answer2;
    private String answer3;
    private String answer4;
    private int correctAnswer; // (1, 2, 3 ou 4)

    public Question() {
    }

    public Question(int testId, String enonce, String answer1, String answer2,
                    String answer3, String answer4, int correctAnswer) {
        this.testId = testId;
        this.enonce = enonce;
        this.answer1 = answer1;
        this.answer2 = answer2;
        this.answer3 = answer3;
        this.answer4 = answer4;
        this.correctAnswer = correctAnswer;
    }

    public Question(int id, int testId, String enonce, String answer1, String answer2,
                    String answer3, String answer4, int correctAnswer) {
        this.id = id;
        this.testId = testId;
        this.enonce = enonce;
        this.answer1 = answer1;
        this.answer2 = answer2;
        this.answer3 = answer3;
        this.answer4 = answer4;
        this.correctAnswer = correctAnswer;
    }

    public int getId() {
        return id;
    }

    public int getTestId() {
        return testId;
    }

    public void setTestId(int testId) {
        this.testId = testId;
    }

    public String getEnonce() {
        return enonce;
    }

    public void setEnonce(String enonce) {
        this.enonce = enonce;
    }

    public String getAnswer1() {
        return answer1;
    }

    public void setAnswer1(String answer1) {
        this.answer1 = answer1;
    }

    public String getAnswer2() {
        return answer2;
    }

    public void setAnswer2(String answer2) {
        this.answer2 = answer2;
    }

    public String getAnswer3() {
        return answer3;
    }

    public void setAnswer3(String answer3) {
        this.answer3 = answer3;
    }

    public String getAnswer4() {
        return answer4;
    }

    public void setAnswer4(String answer4) {
        this.answer4 = answer4;
    }

    public int getCorrectAnswer() {
        return correctAnswer;
    }

    public void setCorrectAnswer(int correctAnswer) {
        this.correctAnswer = correctAnswer;
    }
}
